python_modules/dagster/dagster/core/asset_defs/asset_collection.py

```diff
@@ -16,7 +19,7 @@ class AssetCollection(
         "_AssetCollection",
         [
             ("assets", Sequence[AssetsDefinition]),
-            ("source_assets", Sequence[ForeignAsset]),
+            ("source_assets", Sequence[Union[ForeignAsset, AssetsDefinition]]),
```

Comment:
I see the potential logic in accepting AssetsDefinitions here, but is that necessary for this change? I think this deserves a little more thought.

Reply:
that is fair